plugins/calm-vscode/src/extension.ts

```diff
@@ -0,0 +1,211 @@
+import * as vscode from 'vscode';
+import { languages, Diagnostic, DiagnosticSeverity, Range } from 'vscode';
+import { validate, ValidationOutput, visualize } from '@finos/calm-shared';
+import { JSONPath } from 'jsonpath-plus';
+import * as jsonSourceMap from 'json-source-map';
+import { ValidationOutcome } from '@finos/calm-shared/dist/commands/validate/validation.output';
+
+const CALM_DRAFT_VERSION = '2024-10';
+
+export function activate(context: vscode.ExtensionContext) {
+
+    let panel: vscode.WebviewPanel | undefined = undefined;
+
+    const visualizeCommand = vscode.commands.registerCommand('calm-vscode.visualize', async () => {
+        const editor = vscode.window.activeTextEditor;
+        const columnToShowIn = editor ? editor.viewColumn : undefined;
+
+        if (panel) {
+            panel.reveal(columnToShowIn);
+        } else {
+            panel = vscode.window.createWebviewPanel(
+                'calm-vscode.visualize',
+                'CALM - Visualization',
+                columnToShowIn || vscode.ViewColumn.One,
+                {}
+            );
+
+            //Initial visualization - then will hook into same function for rendering updates.
+            visualizeDocument(editor, panel, columnToShowIn);
+
+            // On update
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            vscode.workspace.onDidSaveTextDocument(async (document) => {
+                if (editor && panel) {
+                    visualizeDocument(editor, panel, columnToShowIn);
+                }
+            });
+
+            panel.onDidDispose(
+                () => {
+                    panel = undefined;
+                },
+                null,
+                context.subscriptions
+            );
+        }
+    });
+
+    const validateCommand = vscode.commands.registerCommand('calm-vscode.validate', () => {
+        const diagnosticCollection = languages.createDiagnosticCollection('calm');
+        vscode.workspace.onDidSaveTextDocument(async (document: vscode.TextDocument) => {
+            if (document.uri.fsPath.endsWith('.json')) {
```

Comment:
not super familiar with building these extensions still so could be wrong - but is this required given the command is configured in the package.json to only work with json files?

Reply:
Fair spot, I added in the logic into the package.json far later.